### PR TITLE
Card vaulting

### PR DESCRIPTION
**Pull request by dbart01**
It's like pole vaulting but with credit cards.

### What this does

Adds support for card vaulting. Introduces a few new classes to make this happen:
- The `Card.Client` is like `Graph.Client` but only for card server
- The `Card.CreditCard` model encapsulates all credit card data and provides an internal dictionary serialization implementation
- Tests with 100% coverage for all the above

### CODE REVIEW

File: Buy/Card.Client.swift [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    /// The `Vault.Client` is a network layer designed to abstract the communication with the Shopify's
+    /// credit card server. For PCI-compliance reasons, this is a standalone service, separate from the
+    /// main `Graph.Client` service.
+    ///
+    public class Client {
+     
+        /// The `URLSession` backing all `Client` network operations. You may provide your own session when initializing a new `Client`.
+        public let session: URLSession
+        
+        internal let apiURL: URL = URL(string: "https://elb.deposit.shopifycs.com/sessions")!
```
Comment (davidmuzi) [inline Buy/Card.Client.swift]: Is this always the same URL? With the old Checkout API, the vault URL would be part of the checkout object

File: Buy/Card.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public struct Card {}
```
Comment (davidmuzi) [inline Buy/Card.swift]: why empty class definition + extension?

File: Buy/Card.Client.swift (hunk 1 above)
Comment (dbart01) [inline Buy/Card.Client.swift]: As per Shopify docs [here](https://help.shopify.com/api/reference/checkout#create_payment_session):
```
The card vault has a static url and is located at https://elb.deposit.shopifycs.com/sessions.
```

File: Buy/Card.swift (hunk 2 above)
Comment (dbart01) [inline Buy/Card.swift]: It's acting as a namespace.

File: Buy/Card.Client.swift (hunk 1 above)
Comment (davidmuzi) [inline Buy/Card.Client.swift]: Thats the admin endpoint.  Are you sure its the same one for Storefront? @maximebedard

File: Buy/Card.swift (hunk 2 above)
Comment (davidmuzi) [inline Buy/Card.swift]: Right - but having the implementation inside the class, in one file, accomplished the same thing

File: Buy/Card.Client.swift (hunk 1 above)
Comment (maximebedard) [inline Buy/Card.Client.swift]: It's the same endpoint (PCI compliant endpoint completely outside of shopify/shopify)

File: Buy/Card.swift (hunk 2 above)
Comment (dbart01) [inline Buy/Card.swift]: Sure, but that pattern doesn't scale. We use exactly the same approach with the `Graph` namespace. The pattern is designed to explicitly declare a namespace - in a separate file. And provide extensions to that namespace for any type that should belong to it.

File: Buy/Card.Client.swift (hunk 1 above)
Comment (dbart01) [inline Buy/Card.Client.swift]: Should we query the `vaultUrl` on `Checkout` instead actually?

Comment (maximebedard) [inline Buy/Card.Client.swift]: I'd say yes, but technically it's always the same 😄

File: Buy/Card.swift (hunk 2 above)
Comment (davidmuzi) [inline Buy/Card.swift]: My bad, I glanced over the `CreditCard` struck _inside_ `Card` 👍

**Pull request merged**